core/src/main/java/software/amazon/awssdk/auth/DefaultCredentialsProvider.java

```diff
@@ -64,13 +66,18 @@ private DefaultCredentialsProvider(Builder builder) {
      */
     private static AwsCredentialsProviderChain createChain(Builder builder) {
         AwsCredentialsProvider[] credentialsProviders = new AwsCredentialsProvider[] {
-                new SystemPropertyCredentialsProvider(),
-                new EnvironmentVariableCredentialsProvider(),
-                new ProfileCredentialsProvider(),
-                ElasticContainerCredentialsProvider.builder()
-                                                   .asyncCredentialUpdateEnabled(builder.asyncCredentialUpdateEnabled)
-                                                   .build(),
-                InstanceProfileCredentialsProvider.builder()
+            new SystemPropertyCredentialsProvider(),
+            new EnvironmentVariableCredentialsProvider(),
+            new ProfileCredentialsProvider(),
+            ContainerCredentialsProvider.builder()
+                                        .credentialsEndpointProvider(new EcsCredentialsEndpointProvider())
+                                        .asyncCredentialUpdateEnabled(builder.asyncCredentialUpdateEnabled)
+                                        .build(),
+            ContainerCredentialsProvider.builder()
+                                        .credentialsEndpointProvider(new FullUriGenericCredentialsEndpointProvider())
+                                        .asyncCredentialUpdateEnabled(builder.asyncCredentialUpdateEnabled)
+                                        .build(),
+            InstanceProfileCredentialsProvider.builder()
                                                   .asyncCredentialUpdateEnabled(builder.asyncCredentialUpdateEnabled)
                                                   .build()
```

Comment:
Is it possible for us to merge the instance profile credentials provider with the container credentials provider?

Reply:
Nitpick: alignment is off here.

Reply:
Fixed